src/mlpack/methods/ann/convolution_rules/fft_convolution.hpp

```diff
@@ -64,23 +64,23 @@ class FFTConvolution
     // Pad filter and input to the output shape.
     filterPadded.resize(inputPadded.n_rows, inputPadded.n_cols);
 
-    output = arma::real(ifft2(arma::fft2(inputPadded) % arma::fft2(
+    arma::Mat<eT> temp = arma::real(ifft2(arma::fft2(inputPadded) % arma::fft2(
         filterPadded)));
 
     // Extract the region of interest. We don't need to handle the padLastDim in
     // a special way we just cut it out from the output matrix.
-    output = output.submat(filter.n_rows - 1, filter.n_cols - 1,
+    output = temp.submat(filter.n_rows - 1, filter.n_cols - 1,
         input.n_rows - 1, input.n_cols - 1);
```

Comment:
I don't know if the switch to a `temp` matrix makes any difference here.

Reply:
It doesn't, but then it prevents us from changing the `output` matrix's dimensions, and hence we can directly pass matrix slices (preventing copies) like below.

Reply:
Actually, the size of the passed matrix changes on assignment of the element wise operation above. But since we're directly passing slices, that can't be possible and gives a compilation error, and so the `temp` matrix is used here.